grails-app/services/org/pih/warehouse/invoice/InvoiceService.groovy

```diff
@@ -460,4 +469,50 @@ class InvoiceService {
         }
         return csv
     }
+
+    CSVPrinter getInvoiceItemsCsv(List<InvoiceItem> invoiceItems) {
+        def g = Holders.grailsApplication.mainContext.getBean('org.grails.plugins.web.taglib.ApplicationTagLib')
```

Comment:
Maybe it is worth to use move this to a function (like it is already done in some places)
```
    def getApplicationTagLib() {
        return Holders.grailsApplication.mainContext.getBean(ApplicationTagLib)
    }
```